src/player.js

```diff
@@ -157,6 +158,7 @@ export default class Player extends FakeEventTarget {
    */
   static runCapabilities(): void {
     Player._logger.debug("Running player capabilities");
+    Player._engines = EngineProvider.getEngines();
```

Comment:
You should init this in the` _engines` static initializer, and remove the importing of `html5` from here(both require and setting it on the `_engines` array

Reply:
rethinking this, maybe don't save the `_engines` in here at all and always call `enginesProvider.getEngines` when wanting to access an engine, also have the current selected `_engine' managed by it so you can have `enginesProvider._selectedEngine` saved on the `enginesProvider` and then you can access it via `enginesProvider.getEngine` etc...

Reply:
i'll remove the _engines from the player - as for the rest of the logic - i'll open a ticket for it for next version -  i also agree with it.